fds/services/dvc_service.py

```diff
@@ -253,12 +254,37 @@ def commit(self, auto_confirm: bool) -> Any:
         execute_command(commit_cmd, capture_output=False)
 
     @staticmethod
-    def push(remote: str) -> Any:
+    def __execute_push_command(remote: str) -> CompletedProcess:
         push_cmd = ["dvc", "push"]
+        # We will only retry once
         if remote:
             push_cmd.append("-r")
             push_cmd.append(remote)
-        execute_command(push_cmd, capture_output=False)
+        return execute_command(
+            push_cmd,
+            capture_output=False,
+            capture_output_and_write_to_stdout=True
+        )
+
+    def push(self, remote: str) -> Any:
+        push_output_bytes = self.__execute_push_command(remote)
+        # Check if its unauthorized
+        if '401 Unauthorized' in convert_bytes_to_string(push_output_bytes.stderr):
+            self.printer.warn("Please enter your credentials, so we can pull from your dvc remote")
+            user_name = get_input_from_user("Enter your dvc username")
+            password = get_input_from_user("Enter your dvc password", type="password")
```

Comment:
Would be a good idea to check if the remote is pointing at DAGsHub, and if so:
1. Change the wording to DAGsHub username and token
2. Point the user to `https://dagshub.com/user/settings/tokens` to get the token instead of a password